### PR TITLE
Add support for external WebIDs registering with username & password

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
(Note: changes default templates & views, be sure to clear `config/views` & `config/templates` before testing.)

### CODE REVIEW

**Comment by RubenVerborgh** [comment]
### Logging in without an account
> This server is not the authorized provider for Web ID https://ruben.verborgh.org/profile/#me. See https://github.com/solid/webid-oidc-spec#authorized-oidc-issuer-discovery

Nice!

### Registering with a garbage WebID

Tried registering with "xyz" as a WebID, it let me… Might be good to add a URI check for people who don't know what WebID is (and might fill out things like their username, "none", etc.).

### Registering with an actual WebID

Works fine; I can login with username as well as WebID, and I get the proper error message when the triple is not set in my profile.

File: default-templates/new-account/index.html [hunk 1]
```diff
@@ -14,10 +14,10 @@
   <div class="row">
     <div class="col-md-12">
       <p style="margin-top: 3em; margin-bottom: 3em;">
-        Welcome to your Solid user profile.
+        Welcome.
       </p>
       <p>
-        Your Web ID is:<br />
+        Web ID:<br />
```
Comment (RubenVerborgh) [inline default-templates/new-account/index.html]: Would it make sense to make this more descriptive? As in `The owner of this page/space is {{webId}}`? There was some confusion in the past with "Your Web ID is", and this removes some confusion, but maybe not all.

File: default-templates/new-account/profile/card [hunk 2]
```diff
@@ -18,6 +18,7 @@
     solid:account </> ;  # link to the account uri
     pim:storage </> ;    # root storage
 
+    solid:inbox </inbox/> ;
```
Comment (RubenVerborgh) [inline default-templates/new-account/profile/card]: Do we need both?

File: lib/handlers/error-pages.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ const SELECT_PROVIDER_AUTH_METHODS = ['oidc']
  * @param next {Function}
  */
 function handler (err, req, res, next) {
-  debug('Error page because of ' + err.message)
+  debug('Error page because of:', err)
```
Comment (RubenVerborgh) [inline lib/handlers/error-pages.js]: Can debug handle a second argument?

File: lib/models/account-manager.js [hunk 4]
```diff
@@ -346,16 +346,32 @@ class AccountManager {
...
+      } else {
+        throw err
       }
+    }
 
-      userConfig.username = this.usernameFromWebId(userConfig.webId)
+    if (userConfig.username) {
+      if (userConfig.externalWebId && !userConfig.localAccountId) {
+        // External Web ID exists, derive the local account id from username
+        userConfig.localAccountId = this.accountWebIdFor(userConfig.username)
```
Comment (RubenVerborgh) [inline lib/models/account-manager.js]: So this means I can automatically get a username assigned?

File: default-templates/new-account/index.html (hunk 1 above)
Comment (dmitrizagidulin) [inline default-templates/new-account/index.html]: Yeah, good idea

File: lib/handlers/error-pages.js (hunk 3 above)
Comment (dmitrizagidulin) [inline lib/handlers/error-pages.js]: Yeah, it's pretty great, like console.log

File: lib/models/account-manager.js (hunk 4 above)
Comment (dmitrizagidulin) [inline lib/models/account-manager.js]: Sort of. the localAccountId just maps from the local server's username-based url to the external Web ID

**Comment by dmitrizagidulin** [comment]
Whoops, I forgot this PR was still unmerged.

File: default-templates/new-account/profile/card (hunk 2 above)
Comment (dmitrizagidulin) [reply default-templates/new-account/profile/card]: Good point, we do not.

**Pull request merged**